dspace-api/src/test/java/org/dspace/content/RelationshipServiceImplTest.java

```diff
@@ -0,0 +1,287 @@
+        relationshipTest.add(getRelationship(cindy, hank, hasFather,0,0));
+        relationshipTest.add(getRelationship(fred, cindy, hasMother,0,0));
+        relationshipTest.add(getRelationship(bob, cindy, hasMother,1,0));
+        when(relationshipService.findByItem(context, cindy)).thenReturn(relationshipTest);
+        when(relationshipDAO.findByItem(context, cindy)).thenReturn(relationshipTest);
+
+
+        List<Relationship> results = relationshipService.findByItem(context, cindy);
+        assertEquals("TestFindByItem 0", relationshipTest, results);
+    }
+
+    @Test
+    public void testFindLeftPlaceByLeftItem() throws Exception {
+        Item item = mock(Item.class);
+        when(relationshipDAO.findLeftPlaceByLeftItem(context, item)).thenReturn(0);
+        assertEquals("TestFindLeftPlaceByLeftItem 0", relationshipDAO.findLeftPlaceByLeftItem(context, item),
+                relationshipService.findLeftPlaceByLeftItem(context, item));
+    }
+
+    @Test
+    public void testFindRightPlaceByRightItem() throws Exception {
+        Item item = mock(Item.class);
+        when(relationshipDAO.findRightPlaceByRightItem(context, item)).thenReturn(0);
+        assertEquals("TestFindRightPlaceByRightItem 0", relationshipDAO.findRightPlaceByRightItem(context, item),
+                relationshipService.findRightPlaceByRightItem(context, item));
+    }
+
+    @Test
+    public void testFindByItemAndRelationshipType() throws Exception {
+        List<Relationship> relList = new LinkedList<>();
+        Item item = mock(Item.class);
+        RelationshipType testRel = new RelationshipType();
+
+        assertEquals("TestFindByItemAndRelationshipType 0", relList,
+                relationshipService.findByItemAndRelationshipType(context, item, testRel, true));
+        assertEquals("TestFindByItemAndRelationshipType 1", relList,
+                relationshipService.findByItemAndRelationshipType(context, item, testRel));
+    }
+
+    @Test
+    public void testFindByRelationshipType() throws Exception {
+        List<Relationship> relList = new LinkedList<>();
+        RelationshipType testRel = new RelationshipType();
+
+        assertEquals("TestFindByRelationshipType 0", relList,
+                relationshipService.findByRelationshipType(context, testRel));
+        assertEquals("TestFindByRelationshipType 1", relList,
+                relationshipService.findByRelationshipType(context, testRel));
+    }
+
+    @Test
+    public void find() throws Exception {
+        Relationship relationship = new Relationship();
+        relationship.setId(1337);
+        when(relationshipDAO.findByID(context, Relationship.class, relationship.getID())).thenReturn(relationship);
+        assertEquals("TestFind 0", relationship, relationshipService.find(context, relationship.getID()));
+    }
+
+    @Test
+    public void testCreate() throws Exception {
+        Relationship relationship = relationshipDAO.create(context,new Relationship());
```

Comment:
Again inline comments would be helpful here. There's a lot of content & relationships getting mocked for this test.